client/app/components/SettingsWrapper.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import Menu from 'antd/lib/menu';
+import { PageHeader } from '@/components/PageHeader';
+import { $location } from '@/services/ng';
+import settingsMenu from '@/services/settingsMenu';
+
+
+function wrapSettingsTab(options, WrappedComponent) {
+  if (options) {
+    settingsMenu.add(options);
+  }
+
+  return function SettingsTab(props) {
+    const activeItem = settingsMenu.getActiveItem($location.path());
+    return (
+      <div className="settings-screen">
+        <div className="container">
+          <PageHeader title="Settings" />
+          <div className="bg-white tiled">
+            <Menu selectedKeys={[activeItem && activeItem.title]} mode="horizontal">
+              {settingsMenu.items.map(item => (
+                <Menu.Item key={item.title}><a href={item.path}>{item.title}</a></Menu.Item>
+              ))}
+            </Menu>
+            <div className="p-15">
+              <div>
+                <WrappedComponent {...props} />
```

Comment:
Very nice :+1:

Haven't you thought about implementing this not as HOC, but as a regular component and then use it to wrap a return value in `render()` functions where needed. Like:

```javascript
function SettingsScreen({ children }) {
  return (
    <div className="settings-screen">
      ...
      {children}
    </div>
  );
}

// Later in other components:

class DataSourcesList extends ReactComponent {
  ...

  render() {
    return (
      <SettingsScreen>
        (Everything that was here before)
      </SettingsScreen>
    );
  }
}
```

I don't mean that there's something wrong with your solution, just wondering why you decided to implement this as HOC :slightly_smiling_face:

Reply:
I did think about it, my reason was to centralize the "registration" of settings tabs in one place (which ended up not being as much as I initially thought :sweat_smile:).

Reply:
Yesh, I didn't notice that first, but now I see. Very cool, I like it :+1:

Reply:
I had the same thought as @kravets-levko :)

Reply:
What I first had in mind was to move the routing part to the HOC function as well, but ended up moving only the `settingsMenu.add`. Anyway, it seemed to make sense to have this logic centralized in there, so I didn't bother to change.

Reply:
There is no much sense to do anything with routing until we replave `angular-router` with React alternative. And even in this case routing should not be coupled with this settings screen HOC. So I think it's good that you decided to not change routing part here